lib/db/mysql.js

```diff
@@ -1490,9 +1498,36 @@ module.exports = function (log, error) {
       })
   }
 
-  const CONSUME_RECOVERY_CODE = 'CALL consumeRecoveryCode_1(?, ?)'
+  const CONSUME_RECOVERY_CODE = 'CALL consumeRecoveryCode_2(?, ?)'
+  const RECOVERY_CODES = 'CALL recoveryCodes_1(?)'
   MySql.prototype.consumeRecoveryCode = function (uid, code) {
-    return this.readFirstResult(CONSUME_RECOVERY_CODE, [uid, dbUtil.createHashSha512(code)])
+    // Consuming a recovery code is done in a two step process because
+    // the stored scrypt hash will need to be calculated against the recovery
+    // code salt.
+    return this.read(RECOVERY_CODES, [uid])
```

Comment:
Can we use `this.readOneFromFirstResult` here to avoid having to dereference `results[0]` below?